org.eclipse.buildship.core/src/main/java/org/eclipse/buildship/core/workspace/internal/SynchronizeGradleBuildOperation.java

```diff
@@ -244,6 +244,7 @@ private void synchronizeJavaProjectInTransaction(final OmniEclipseProject projec
         JavaSourceSettingsUpdater.update(javaProject, project, progress.newChild(1));
         GradleClasspathContainerUpdater.updateFromModel(javaProject, project, SynchronizeGradleBuildOperation.this.allProjects, persistentModel, progress.newChild(1));
         WtpClasspathUpdater.update(javaProject, project, progress.newChild(1));
+        CorePlugin.externalLaunchConfigurationManager().updateClasspathProviders(workspaceProject);
```

Comment:
⭕️ This seems unrelated to this change. Why is it here?

Reply:
This is actually related to this change and covers a missing use-case. The Gradle classpath provider is updated upon certain events (Gradle nature added, project created, etc.). The update only considers Java projects. Now, we create a generic root project and add the Java nature later, there's a chance that some run configurations are not updated properly. 

Calling `updateClasspathProviders()` at the end of the synchronization adds minimal overhead and ensures that no matter how we adjust the project configuration, the classpath providers will be up-to-date after a project synchronization.